sugondat-shim/src/cmd/query/block.rs

```diff
@@ -0,0 +1,43 @@
+use super::connect_rpc;
+use crate::cli::query::block::{BlockRef, Params};
+
+pub async fn run(params: Params) -> anyhow::Result<()> {
+    let Params { rpc, block } = params;
+
+    let client = connect_rpc(rpc).await?;
+
+    let maybe_hash = match block {
+        BlockRef::Best => None,
+        BlockRef::Hash(h) => Some(h),
+        BlockRef::Number(n) => Some(
+            client
+                .block_hash(n)
```

Comment:
Just a note: so far commands behave a lot like adapters and those will wait until the given block is available. Do you think this is appropriate here?

Reply:
No, I'd overlooked that. I would have expected the `block_hash` function to bail early if no such block exists, and rather have something like a `wait_for_block_hash` for the current behavior. Let's address it in a follow-up.

Reply:
Or do you mean that it _should_ wait until that block is available?

If so, we might introduce a CLI flag - `--wait`, or `--no-wait`, depending on the default desired behavior. I don't have a strong opinion - let's discuss

Reply:
yeah, or more precise, I am wondering if it should wait until that block is available. I think a flag would make sense.